src/compiler/crystal/semantic/call_error.cr

```diff
@@ -391,13 +391,11 @@ class Crystal::Call
 
   def append_matches(defs, arg_types, str, *, matched_def = nil, argument_name = nil)
     defs.each do |a_def|
-      str << "\n - "
-      append_def_full_name a_def.owner, a_def, arg_types, str
+      str << "\n - #{def_full_name(a_def.owner, a_def, arg_types)}".colorize.bold
```

Comment:
With this change, I suppose the `append_def_full_name` method(s) can be removed in favour of `def_full_name`? They're no longer used anywhere. At the very least the instance method.

Reply:
I didn't remove them because `def_full_name` actually depends on `append_def_full_name` because it's used to build the string returned by `def_full_name`.
But I could of course put `append_def_full_name`'s code directly into the `String.build { |io| }` of `def_full_name` but I'm not sure if that'd be a good readability improvement.

Reply:
I don't know why these are defined as both instance and class methods anyway. Clearly, they're not bound to the state of the object, but they're not used from outside the class either. Maybe it's left over. I guess this could all need a refactoring, but not in this PR.